online_inference/make_request.py

```diff
@@ -0,0 +1,55 @@
+import sys
+import numpy as np
+import pandas as pd
+import requests
+import click
+import logging
+import urllib
+import json
+from app_predict import XInput, YResponse
+
+logger = logging.getLogger(__name__)
+handler = logging.StreamHandler(sys.stdout)
+log_fmt = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+logging.basicConfig(level=logging.INFO, format=log_fmt)
+
+
+def make_predict(server_address: str,
+                 path_to_data: str,
+                 path_for_save_result: str):
+    logger.info(f"read data")
+    data = pd.read_csv(path_to_data)
+    data_for_predict = XInput(data=data.values.tolist(), features=data.columns.tolist())
+    logger.info(f"send request")
+    response = requests.get(
+        urllib.parse.urljoin(server_address, "predict"),
+        json=data_for_predict.dict(),
+    )
+    logger.info(f"parse result")
+    result = []
+    if (response.status_code == 200):
+        for elem in response.json():
+            predict = YResponse(**elem)
+            result.append(predict.predict)
+
+        logger.info(f"save result")
+        with open(path_for_save_result, "w") as result_file:
+            json.dump(result, result_file)
+    else:
+        logger.info(f"save result")
+        with open(path_for_save_result, "w") as result_file:
+            json.dump(response.json(), result_file)
```

Comment:
Тут можно было бы обойтись без лишней копипасты:

```suggestion
    else:
        result = response.json()
    logger.info(f"save result")
    with open(path_for_save_result, "w") as result_file:
        json.dump(result, result_file)
```